code/modules/chemistry/tools/dispensers.dm

```diff
@@ -464,38 +464,46 @@
 	icon_state = "still"
 	amount_per_transfer_from_this = 25
 	event_handler_flags = NO_MOUSEDROP_QOL
-	// what was the point here exactly
-	//New()
-		//..()
+	var/list/random_reagents_list
 
 	proc/brew(var/obj/item/W as obj)
-		if (!W || !(istype(W,/obj/item/reagent_containers/food) || istype(W, /obj/item/plant)))
-			return 0
+		var/brewable
+		var/list/brew_result
+
+		if(istype(W,/obj/item/reagent_containers/food))
+			var/obj/item/reagent_containers/food/F = W
+			brewable = F.brewable
+			brew_result = F.brew_result
+
+			if (istype(W,/obj/item/reagent_containers/food/snacks/snack_cake) || istype(W,/obj/item/reagent_containers/food/snacks/burrito))
+				if(islist(brew_result))
+					brew_result += F.reagents.reagent_list
+
+		else if(istype(W, /obj/item/plant))
+			var/obj/item/plant/P = W
+			brewable = P.brewable
+			brew_result = P.brew_result
 
-		if (!W:brewable || !W:brew_result)
+		if (!brewable || !brew_result)
 			return 0
 
-		if (istype(W,/obj/item/reagent_containers/food/snacks/snack_cake) || istype(W,/obj/item/reagent_containers/food/snacks/burrito))
-			if (islist(W:brew_result) && W:brew_result:len)
-				W:brew_result += W.reagents.reagent_list
-
-		//var/brewed_name = null
-		if (islist(W:brew_result) && W:brew_result:len)
-			for (var/i in W:brew_result)
-				//brewed_name += ", [reagent_id_to_name(i)]"
-				src.reagents.add_reagent(i, 10)
-			//brewed_name = copytext(brewed_name, 3)
+		if (islist(brew_result) && length(brew_result))
+			for (var/i in brew_result)
+				src.reagents.add_reagent(i, 20) // Why has it always been like this omegaweed makes a shitton of chem
+				if(length(src.random_reagents_list)) // should this be an A or B not A AND B?
+					src.reagents.add_reagent(pick(random_reagents_list), 20/length(brew_result))
```

Comment:
So these comments were my: "I don't understand what was intended because it seems problematic and here is why." I should have annotated them accordingly. 😊 

See Pali's [comment](https://github.com/goonstation/goonstation/pull/3406#issuecomment-834160836).

Also consider rounding the number of reagents to generate.